Modified puppet module to include bool for using no-use_srv_records flag

brettmilford opened the pull request:
##### SUMMARY
This change adds an option to toggle the --no-use_srv_records flag in a puppet agent run command with the puppet module

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Puppet module no-use_srv_records flag

##### ADDITIONAL INFORMATION

brettmilford committed:
Modified puppet module to include bool for using no-use_srv_records flag

brettmilford committed:
removed version_added, flaged by validate-modules test

brettmilford committed:
fixed version_added as flagged by lint

resmo commented on lib/ansible/modules/system/puppet.py:
```diff
@@ -192,6 +197,7 @@ def main():
             summarize=dict(type='bool', default=False),
             debug=dict(type='bool', default=False),
             verbose=dict(type='bool', default=False),
+            no_use_srv_records=dict(type='bool', default=False),
```
I would change it in a way to toggle either`--use_srv_records` or --no-use_srv_records` whether this value is true or false

~~~
 no_use_srv_records=dict(type='bool'),

resmo commented on lib/ansible/modules/system/puppet.py:
```diff
@@ -260,6 +266,8 @@ def main():
             cmd += " --certname='%s'" % p['certname']
         if module.check_mode:
             cmd += " --noop"
+        if p['no_use_srv_records']:
```
~~~python
        if p['use_srv_records'] is not None:
            if not p['use_srv_records']:
                cmd += " --no-use_srv_records"
            else:
                cmd += " --use_srv_records"
~~~

does this make sense?

brettmilford replied:
It makes perfect sense, I did originally set out to do that, but couldn't find any documentation on the `--use_srv_recrods` flag.

resmo commented on lib/ansible/modules/system/puppet.py:
```diff
@@ -82,6 +82,11 @@
       - It has no effect with a puppetmaster.
     type: str
     version_added: "2.1"
+  no_use_srv_records:
```
use_srv_records:

resmo reviewed (commented):
needs_info

brettmilford committed:
Reworked use_srv_records option

Now adds the use_srv_records flag or no-use_srv_records flag based on the value
of use_srv_records opt.

resmo commented on lib/ansible/modules/system/puppet.py:
```diff
@@ -260,6 +266,11 @@ def main():
             cmd += " --certname='%s'" % p['certname']
         if module.check_mode:
             cmd += " --noop"
+        if p['use_srv_records']:
```
this should be validaidating to non None:

~~~python
if p['use_srv_records'] is not None:
~~~

brettmilford committed:
fixed conditional validation

brettmilford commented:
@resmo Is there some additional information or other action required to merge this PR?
Thanks.
bot_status

brettmilford commented:
bot_status

resmo committed:
adjust version added

resmo reviewed (approved):
shipit

resmo commented:
Thanks for pinging, I adjusted the version_added to 2.9 and wait for the build before merge. Thanks again!
